### PR TITLE
change: [M3-6522] - Tag Component v7 story & cleanup

### PR DESCRIPTION
**Pull request by abailly-akamai**
## Description 📝
This PR replaces the Tag component's `.mdx` story with a fresh v7 story.

## Changes  🔄
- Refactor old mdx story into a v7 story
- Component cleanup
- A new .styles file
- A new unit test
- A few fixes:

**Broken hover state in main menu** 
| Before  | After   |
| ------- | ------- |
| Screenshot 2023-10-24 at 16 26 24 | Screenshot 2023-10-24 at 16 43 36 |

**Poor contrast & accessibility on dark mode**
| Before  | After   |
| ------- | ------- |
| Screenshot 2023-10-24 at 16 42 57 | Screenshot 2023-10-24 at 16 47 22 |

## Preview 📷
Apart from the fixes shown above, there should be no change in the way the component looks or behaves.

### Prerequisites
- Pull code locally

### Verification steps 
(How to verify changes)
- Navigate to http://localhost:6006/?path=/docs/components-chip-tag--documentation to confirm the new story
- Check the tag in CM to ensure no regression and the small fixes (main search with tagged linodes, tag panels in entity details)

## As an Author I have considered 🤔

*Check all that apply*

- [x] 👀 Doing a self review
- [x] ❔ Our [contribution guidelines](https://github.com/linode/manager/blob/develop/docs/CONTRIBUTING.md)
- [ ] 🤏 Splitting feature into small PRs
- [x] ➕ Adding a changeset
- [x] 🧪 Providing/Improving test coverage
- [ ] 🔐 Removing all sensitive information from the code and PR description
- [ ] 🚩 Using a feature flag to protect the release
- [x] 👣 Providing comprehensive reproduction steps
- [ ] 📑 Providing or updating our documentation
- [ ] 🕛 Scheduling a pair reviewing session
- [x] 📱 Providing mobile support
- [x] ♿  Providing accessibility support

### CODE REVIEW

File: packages/manager/src/components/Tag/Tag.tsx [hunk 1]
```diff
@@ -1,155 +1,92 @@
...
 import { truncateEnd } from 'src/utilities/truncate';
 
+import { StyledChip, StyledDeleteButton } from './Tag.styles';
+
+import type { ChipProps } from 'src/components/Chip';
+
 type Variants = 'blue' | 'lightBlue';
 
 export interface TagProps extends ChipProps {
-  asSuggestion?: boolean;
```
Comment (abailly-akamai) [inline packages/manager/src/components/Tag/Tag.tsx:13]: This was only used as a boolean for `closeMenu` (in main search bar) so i took it out and checked for `closeMenu` directly

File: packages/manager/src/components/Tag/Tag.tsx [hunk 2]
```diff
@@ -1,155 +1,92 @@
...
+    if (closeMenu) {
+      closeMenu();
     }
     history.push(`/search/?query=tag:${label}`);
   };
 
   // If maxLength is set, truncate display to that length.
   const _label = maxLength ? truncateEnd(label, maxLength) : label;
 
-  const tagProps = omit(props, ['asSuggestion', 'closeMenu']);
```
Comment (abailly-akamai) [inline packages/manager/src/components/Tag/Tag.tsx:38]: we can just use shouldForwardProps on the styled component for this

**Review by bnussman-akamai — APPROVED** [review]

File: packages/manager/src/components/Tag/Tag.test.tsx [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+
+  it('truncates the label if maxLength is provided', () => {
+    const { getByText } = renderWithTheme(
+      <Tag {...defaultProps} label="Long Label" maxLength={5} />
+    );
+    const tagElement = getByText('Lo...');
+    expect(tagElement).toBeInTheDocument();
+  });
+
+  it('calls closeMenu and navigates when clicked', () => {
```
[the remark this answers is not in the document]
Comment (abailly-akamai) [reply packages/manager/src/components/Tag/Tag.test.tsx]: nope! removed

**Review by jdamore-linode — APPROVED** [review]
Thanks @abailly-akamai! Changes look good in Storybook, and I'm always excited to see doc improvements and test additions

**Pull request merged**